homeassistant/components/ecodevices/sensor.py

```diff
@@ -0,0 +1,263 @@
+"""Support for the GCE Eco-Devices."""
+import logging
+
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    CONF_C1_DEVICE_CLASS,
+    CONF_C1_ENABLED,
+    CONF_C1_NAME,
+    CONF_C1_UNIT_OF_MEASUREMENT,
+    CONF_C2_DEVICE_CLASS,
+    CONF_C2_ENABLED,
+    CONF_C2_NAME,
+    CONF_C2_UNIT_OF_MEASUREMENT,
+    CONF_T1_ENABLED,
+    CONF_T1_NAME,
+    CONF_T1_UNIT_OF_MEASUREMENT,
+    CONF_T2_ENABLED,
+    CONF_T2_NAME,
+    CONF_T2_UNIT_OF_MEASUREMENT,
+    CONFIG,
+    CONTROLLER,
+    COORDINATOR,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the GCE Eco-Devices platform."""
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    controller = data[CONTROLLER]
+    coordinator = data[COORDINATOR]
+    config = data[CONFIG]
+
+    entities = []
+
+    if config.get(CONF_T1_ENABLED):
+        _LOGGER.debug("Add the teleinfo 1 entity")
+        entities.append(
+            T1EdDevice(
+                controller,
+                coordinator,
+                "t1",
+                config.get(CONF_T1_NAME),
+                config.get(CONF_T1_UNIT_OF_MEASUREMENT),
+                "power",
```

Comment:
```suggestion
                DEVICE_CLASS_POWER,
```

from homeassistant.const